### PR TITLE
Spawn nodes for cluster

### PR DESCRIPTION
**Pull request by robertgzr**
Adds commands to bring cluster nodes up and down, list running nodes and run kubeadm

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,6 +8,144 @@ on a single machine, created mostly for developers __of__ Kubernetes.
...
+```
+
+Sometimes when you Docker doesn't use the newest existing API, you may see
+the following error:
+
+```
+2017/01/26 16:41:38 Error when pushing image: Error response from daemon: client is newer than server (client API version: 1.26, server API version: 1.24)
+```
+
+Then wou will need to include your Docker API version in DOCKER_API_VERSION
```
Comment (alban) [inline README.md]: s/wou/you

File: pkg/distribution/registry.go [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+		if errMsg, ok := jsonLine["error"]; ok {
+			if errMsgStr, ok := errMsg.(string); ok {
+				return fmt.Errorf(errMsgStr)
+			}
+		}
+
+		if progress, ok := jsonLine["progress"]; ok {
+			fmt.Println(progress)
+		}
+		// fmt.Println(string(line[:]))
```
Comment (nhlfr) [inline pkg/distribution/registry.go]: needs to be uncommented

File: README.md [hunk 3]
```diff
@@ -8,6 +8,144 @@ on a single machine, created mostly for developers __of__ Kubernetes.
...
+In the directory where you cloned this repository, please do:
+
+```
+make
+sudo GOPATH=$GOPATH SYSTEMD_NSPAWN_PATH=<path_to_your_nspawn_binary> IMAGE_METHOD=mkosi ./kubeadm-systemd up --nodes <number_of_nodes>
+sudo ./kubeadm-systemd init
+sudo ./kubeadm-systemd down
+```
+
+Sometimes when you Docker doesn't use the newest existing API, you may see
```
Comment (alban) [inline README.md]: Do you nean `s/you //`?

File: pkg/cnispawn/spawn.go [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+			return err
+		}
+	}
+
+	// TODO(nhlfr): Don't hardcode kubeadm-systemd specific options,
+	// expose all the options of systemd-nspawn instead.
+	args := []string{
+		systemdNspawnPath,
+		"--capability=cap_audit_control,cap_audit_read,cap_audit_write,cap_audit_control,cap_block_suspend,cap_chown,cap_dac_override,cap_dac_read_search,cap_fowner,cap_fsetid,cap_ipc_lock,cap_ipc_owner,cap_kill,cap_lease,cap_linux_immutable,cap_mac_admin,cap_mac_override,cap_mknod,cap_net_admin,cap_net_bind_service,cap_net_broadcast,cap_net_raw,cap_setgid,cap_setfcap,cap_setpcap,cap_setuid,cap_sys_admin,cap_sys_boot,cap_sys_chroot,cap_sys_module,cap_sys_nice,cap_sys_pacct,cap_sys_ptrace,cap_sys_rawio,cap_sys_resource,cap_sys_time,cap_sys_tty_config,cap_syslog,cap_wake_alarm",
+		// "--bind=/proc/sys/net/bridge",
```
Comment (nhlfr) [inline pkg/cnispawn/spawn.go]: needs to be removed

File: cmd/kubeadm-systemd/kubeadm-systemd.go [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+	for i, node := range nodes {
+		if i != 0 {
+			if err := ssh.JoinNode(node.IP.String(), token, nodes[0].IP.String()); err != nil {
+				if err := nspawntool.Cleanup(len(nodes)); err != nil {
+					log.Fatal("Error when cleaning up: ", err)
+				}
+				log.Fatal("Error when joining node: ", err)
+			}
+		}
+
```
Comment (alban) [inline cmd/kubeadm-systemd/kubeadm-systemd.go]: empty line

File: scripts/node/root/weave-daemonset.yaml [hunk 6]
```diff
@@ -0,0 +1,75 @@
+apiVersion: extensions/v1beta1
```
Comment (alban) [inline scripts/node/root/weave-daemonset.yaml]: Where does this file "weave-daemonset.yaml" come from? Could you add an URL in a comment?

Comment (robertgzr) [inline scripts/node/root/weave-daemonset.yaml]: not sure where it comes from. maybe @nhlfr knows

File: pkg/cnispawn/spawn.go [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+		// "--capability=cap_audit_write,cap_audit_control,cap_sys_admin,cap_sys_chroot,cap_net_admin,cap_setfcap,cap_syslog",
```
Comment (robertgzr) [inline pkg/cnispawn/spawn.go]: @nhlfr what about this line?

Comment (nhlfr) [inline pkg/cnispawn/spawn.go]: please remove it too

**Comment by nhlfr** [comment]
Can we close this PR?

**Comment by robertgzr** [comment]
I was still waiting on feedback from you and @alban 

If you are alright with it I would like to merge this because it closes at least 2 issues

**Comment by robertgzr** [comment]
I could still split it into 3 commits like you suggest. But if I remember correctly I had some issues with the renaming of `cmd/kubeadm-systemd/kubeadm-systemd`  to `*-nspawn`

**Pull request merged**